peekingduck/pipeline/utils/bbox/transforms.py

```diff
@@ -107,23 +291,58 @@ def xywh2xyxy(inputs: torch.Tensor) -> torch.Tensor:
         tensor([[ 85.0, 180.0, 115.0, 220.0]])
 
     Args:
-        inputs (torch.Tensor): Input bounding boxes (2-d array) each with the
+        inputs (BboxType): Input bounding boxes of shape (..., 4) each with the
             format `(center x, center y, width, height)`.
 
     Returns:
-        (torch.Tensor): Bounding boxes with the format `(top left x, top left y,
-        bottom right x, bottom right y)`.
+        outputs (BboxType): Bounding boxes with the format `(top left x, top left y,
+            bottom right x, bottom right y)`.
+    """
+    outputs = clone(inputs)
+    outputs = cast_int_to_float(outputs)
+
+    outputs[..., 0] = inputs[..., 0] - inputs[..., 2] / 2
+    outputs[..., 1] = inputs[..., 1] - inputs[..., 3] / 2
+    outputs[..., 2] = inputs[..., 0] + inputs[..., 2] / 2
+    outputs[..., 3] = inputs[..., 1] + inputs[..., 3] / 2
+
```

Comment:
Added a new utility function `empty_like` to do `isinstance` check for input type and perform `np.empty_like` or `torch.empty_like` conditionally.

Since my `clone`, `cast_to_float` and now `empty_like` all use the same conditional checking:

```
if isinstance(inputs, torch.Tensor):
    do something using torch api
do something with numpy api
```

Is there a good design (pythonic way) to only check the type once?

Reply:
> Is there a good design (pythonic way) to only check the type once?

I don't really have a good solution for this. It might be possible if you combine everything into `cast_to_float` with addition `empty_like` and `clone` flag to avoid multiple checking. But I'm not sure if that will affect readability.

I think it's fine to leave it as it is first and we can revisit it if we come across a good sample implementation in the future.

Reply:
Noted on this.